examples/version-migration/separate-container/tests/separateContainer.test.ts

```diff
@@ -151,13 +153,13 @@ describe("separate-container migration", () => {
 			});
 
 			const migratorsLength = await page.evaluate(() => {
-				return window["migrators"].length;
+				return (window["migrators"] as IMigrator[]).length;
 			});
-			await expect(migratorsLength).toEqual(2);
+			expect(migratorsLength).toEqual(2);
 
 			// Get a promise that will resolve when both sides have finished migration
-			const migrationP = page.evaluate(() => {
-				const migrationPs = (window["migrators"] as IMigrator[]).map((migrator) => {
+			const migrationP = page.evaluate(async () => {
+				const migrationPs = (window["migrators"] as IMigrator[]).map(async (migrator) => {
 					return new Promise<void>((resolve) => {
```

Comment:
There is a check above for migration status to be "one", but time has passed and we aren't in control of all timing and have questions about puppeteer. You _could_ assert before the return that status is still one just be really defensive.

Reply:
We could change to failing the test, but I don't know that it is worth it.
`assert` could provide a better debugging opportunity... if we generally fix up support to break under debugger to get immediate context instead of delayed until promise is checked.
I would also recommend a positive check that version is exactly as it was before - "one". That handles any funny intermediate state we also don't expect.
Still approve :)